src/environment.cpp

```diff
@@ -681,20 +658,22 @@ typedef struct {
 				continue;
 			v3s16 p = p0 + block->getPosRelative();
 
-			if (!m_aabms[c])
+			if (!m_aabms[c]) {
+				if (block->content_only)
+					return;
 				continue;
+			}
 
 			for(auto & ir: *(m_aabms[c])) {
 				auto i = &ir;
-				if(myrand() % i->chance != 0)
-					continue;
 				// Check neighbors
-				MapNode neighbor;
-				auto & required_neighbors = activate ? ir.abmws->required_neighbors_activate : ir.abmws->required_neighbors;
+				v3POS neighbor_pos;
+				//auto & required_neighbors = activate ? ir.abmws->required_neighbors_activate : ir.abmws->required_neighbors;
```

Comment:
What's with required_neighbors_activate, it doesn't seem to be used anywhere anymore? Should be just deleted then, also `activate` argument of this function isn't used either.